### PR TITLE
chore: use shim::TokenAmount in message related code

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:
- Switch the `MessageTrait` to using `shim::TokenAmount`

## Reference issue to close (if applicable)

Closes 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

**Comment by lemmih** [comment]
This PR is ready for review.

@LesnyRumcajs @elmattic @creativcoder @hanabi1224 @sudo-shashank

File: vm/message/src/chain_message.rs [hunk 1]
```diff
@@ -46,10 +47,10 @@ impl MessageTrait for ChainMessage {
             Self::Unsigned(t) => t.sequence,
         }
     }
-    fn value(&self) -> &TokenAmount {
+    fn value(&self) -> TokenAmount {
         match self {
             Self::Signed(t) => t.value(),
-            Self::Unsigned(t) => &t.value,
+            Self::Unsigned(t) => TokenAmount::from(&t.value),
```
[the remark this answers is not in the document]
Comment (lemmih) [reply vm/message/src/chain_message.rs:53]: `t.value.into()` tries to take ownership of `t`, unfortunately.

[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply vm/message/src/chain_message.rs:53]: `(&t.value).into()`

[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply vm/message/src/chain_message.rs:53]: or `value.borrow().into()`

**Review by elmattic — APPROVED** [review]

**Review by hanabi1224 — APPROVED** [review]

**Comment by elmattic** [comment]
Any reason not to use shim in say `get_base_fee_lower_bound` from `utils` module?

**Pull request merged**